### DISCUSSION TITLE
How to download SQLite dump?

### DISCUSSION THREAD

**Discussion by doaortu** [sqlitecloud/discussions#5] [Feature request] [answered]
Hi, thanks for making this great product.
One thing I miss from SQLite Cloud is ability to download database dump to local, so that I can backup said database locally.
I know there's a feature to backup in SQLite Cloud, but I need a finer control of how I backup my data.
an endpoint to dump the database and download it will be nice.
Thanks.

**Comment by marcobambini** [comment, marked as answer]
Hello @doaortu,
You can download your SQLite database from within the Dashboard.

Go to Databases and select Download from the action menu.
![Screenshot 2024-04-22 at 21 31 19](https://github.com/sqlitecloud/discussions/assets/11838145/9547a647-30c7-4312-834a-7e100055445e)

**Reply by doaortu** [reply]
Whops, I'm sorry, didn't see that little dropdown, because In my laptop screen, apparently I need to scroll to the right to see the dropdown, I thought there's no dropdown this whole time.

Thanks.
But can we do it through API? I want to automate it to download the dump, once a day.

Or  can I use my own s3 storage to store the backup databases?

**Reply by doaortu** [reply]
Hi @marcobambini ,
after I tried to download the db from the dashboard, I noticed that the first download result in corrupt db 
it says : `Error: database disk image is malformed` when I open it in sqlite3 locally
maybe it's because I also have running transaction when I download the db.

Is there any way to prevent that happen while also have running transaction?
Thanks.
